### DISCUSSION TITLE
Problem converting public key generated by `startRegistration` into proper Buffer for passkey authentication

### DISCUSSION THREAD

**Discussion by krisstern** [MasterKale/SimpleWebAuthn#614] [Troubleshooting]
Hi everyone,

I have been trying to convert the public key string (one starting with the letter "M") generated by the `startRegistration` method into a format that is acceptable for passkey authentication by the `verifyAuthenticationResponse`, but it seems the results I got is a `TypeError: cosePublicKey.get is not a function` with `credentialPublicKey: Uint8Array(91)`. I have tried using the test `credentialPublicKey` with value `pQECAyYgASFYILTrxTUQv3X4DRM6L_pk65FSMebenhCx3RMsTKoBm-AxIlggEf3qk5552QLNSh1T1oQs7_2C2qysDwN4r4fCp52Hsqs` as used in "https://github.com/MasterKale/SimpleWebAuthn/blob/master/packages/server/src/authentication/verifyAuthenticationResponse.test.ts", which can be converted in into a `Uint8Array` of size `77`, which seemed to have been deemed acceptable by the `verifyAuthenticationResponse` method as input, though of course as this is dummy data the user is not verified as a result. 

I wish someone with the insight could shed some light on this issue I have encountered. Any help in this regard would be much appreciated.

**Comment by krisstern** [comment]
To provide more context I am using the following config in my `PublicKeyCredentialCreationOptionsJSON` for the input options of the `startRegistration` method:

```typescript
pubKeyCredParams: [
    { alg: -7, type: 'public-key' },
    { alg: -257, type: 'public-key' },
],
```

**Comment by krisstern** [comment]
And, I am using something like the below to convert the publicKey:

```typescript
import { isoBase64URL } from '@simplewebauthn/server/helpers';

const base64URLPublicKey = isoBase64URL.fromBuffer(publicKey);
```

**Comment by Sreejit-Sengupto** [comment]
Yes I have somewhat similar problem, I am using appwrite, and I thought of saving the the 'verificationResultInfo' that contains publicKey in database in string format, by doing JSON.stringify, but the method is altering the verificationResult, what are the ideal ways of doing it? Thanks in advance.

**Comment by kettanaito** [comment]
To whomever is having this, the `TypeError: cosePublicKey.get is not a function` error indicates that your public key isn't correctly encrypted. Depending on your setup, you have to make sure that the public key you provide to `@simplewebauthn` is a valid CORB key. [This](https://github.com/kettanaito/test-passkey/blob/a062487f95605b7623b1550fdabc2d3e22f08948/index.ts#L56-L69) is what I had to do, your mileage may vary.
